libr/core/cmd_print.c

```diff
@@ -2341,7 +2341,9 @@ r_cons_pop();
 		if (asm_flags) {
 			str = strstr (line, ";-- ");
 			if (str) {
-				r_cons_printf ("%s\n", str);
+				if (!(strstr (line, "case") || strstr (line, "switch"))) {
```

Comment:
check with !strncmp because switch and case must be the first word after ";-- ", so we dont miss any false positive then